mods/ra/maps/soviet-02a/soviet02a.lua

```diff
@@ -0,0 +1,170 @@
+		Utils.Do(buildings, function(actor)
+			Trigger.OnDamaged(actor, function(building, attacker)
+				if building.Owner == germany and building.Health < building.MaxHealth * 0.8 then
+					building.StartBuildingRepairs()
+					if attacker.Type ~= "Yak" and not AlreadyHunting then
+						AlreadyHunting = true
+						Utils.Do(germany.GetGroundAttackers(), function(unit)
+							Trigger.OnIdle(unit, unit.Hunt)
+						end)
+					end
+				end
+			end)
+		end)
+
+		-- Find the bridge actors
+		bridgepart1 = Map.ActorsInBox(waypoint23.CenterPosition, waypoint49.CenterPosition, function(self) return self.Type == "br1" end)[1]
+		bridgepart2 = Map.ActorsInBox(waypoint23.CenterPosition, waypoint49.CenterPosition, function(self) return self.Type == "br2" end)[1]
+	end)
+
+	-- Discover the area around the bridge exposing the two german soldiers
+	-- When the two infantry near the bridge are discovered move them accross the bridge to waypoint4
+	-- in the meanwhile one USSR soldier hunts them down
+	Trigger.AfterDelay(DateTime.Seconds(1), function()
+		Actor.Create("camera", true, { Owner = player, Location = waypoint23.Location })
+
+		Utils.Do(FleeingUnits, function(unit)
+			unit.Move(waypoint4.Location)
+		end)
+		Follower.AttackMove(waypoint4.Location)
+	end)
+
+	-- To make it look more smooth we will blow up the bridge when the barrel closest to it blows up
+	Trigger.OnAnyKilled({ BridgeBarrel1, BridgeBarrel2 }, function()
+		-- Destroy the bridge
+		if not bridgepart1.IsDead then
+			bridgepart1.Kill()
+		end
+		if not bridgepart2.IsDead then
+			bridgepart2.Kill()
+		end
+
+		-- Move units towards the oilpump
+		--[[Utils.Do(FleeingUnits, function(unit)
+			if not unit.IsDead then
+				unit.Move(waypoint53.Location)
+			end
+		end)]]
+	end)
+
+	-- If player passes over the bridge, blow up the barrel and destroy the bridge
+	Trigger.OnEnteredFootprint(PassingBridgeLocation, function(unit, id)
+		if unit.Owner == player then
+			Trigger.RemoveFootprintTrigger(id)
+
+			-- Also don't if the bridge is already dead
+			if bridgepart1.IsDead and bridgepart2.IsDead then
+				return
+			end
+
+			-- Don't "shoot" at the barrels if there is noone to shoot
+			if not FleeingUnits[1].IsDead then
+				FleeingUnits[1].Attack(Barrel)
+			elseif not FleeingUnits[2].IsDead then
+				FleeingUnits[2].Attack(Barrel)
+			end
+		end
+	end)
+
+	-- Four infantry from the small island move towards the USSR command center and attack it after 24 Seconds
+	Trigger.AfterDelay(DateTime.Seconds(24),  function()
+		Utils.Do(CmdAtk, function(unit)
+			unit.AttackMove(AttackWaypoint1.Location)
+			Trigger.OnIdle(unit, unit.Hunt)
+			--Trigger.OnIdle(unit, function() unit.AttackMove(AttackWaypoint1.Location) end)
+		end)
+
+		-- On reaching waypoint 7 remove the trigger to keep moving or substitue with new instruction
+		--[[Trigger.OnEnteredFootprint({ AttackWaypoint1.Location }, function(unit, id)
```

Comment:
Similar case here. I cut this down to [two lines](https://github.com/OpenRA/OpenRA/pull/8285/files#diff-b79eef01e62d56a3c8ca44d3db6e74efR113).